### PR TITLE
Get rid of baseURL in IndexFileStream, add tests. fixes #166

### LINKED ISSUE #166: Get rid of IndexFileStream's baseURL code
We're not using base URLs in the index files anymore as of aff1f5c79d319c23a1298847140f234d3a1df167, so we should get rid of the relevant code in `IndexFileStream`. I think it's also our only use of `path-posix` so we can probably get rid of that too.

AND WE'RE GONNA NEED A TEST!

### PR DESCRIPTION
**Pull request by toolness**
@alicoding can you take a look at this?

### CODE REVIEW

File: lib/gulp-index-file-stream.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 var Readable = require('stream').Readable;
-var posixPath = require('path-posix');
```
Comment (alicoding) [inline lib/gulp-index-file-stream.js]: can you remove `path-posix` from package.json too?

Comment (toolness) [inline lib/gulp-index-file-stream.js]: Ah good catch! Done.

File: lib/gulp-index-file-stream.js [hunk 2]
```diff
@@ -22,14 +22,13 @@ IndexFileStream.prototype._read = function() {
...
   var url = this._urls.pop();
-  var indexFile = posixPath.join(
+  var indexFile = path.join(
     this._baseDir,
-    posixPath.join.apply(posixPath, url.split('/').slice(1, -1)),
+    path.join.apply(path, url.split('/').slice(1, -1)),
     'index.html'
   );
 
   this._indexStatic.generate(url, {
```
Comment (alicoding) [inline lib/gulp-index-file-stream.js]: I don't know if we need second parameter here anymore? What do you think @toolness ?

**Comment by alicoding** [comment]
Left one question regarding the removal of `options` object parameter. If you wish to fix then good to land otherwise also good to land. Your call :)

File: lib/gulp-index-file-stream.js (hunk 2 above)
Comment (toolness) [inline lib/gulp-index-file-stream.js]: Yeah, I'm just suspecting we'll actually need options at some point in the future and didn't want to deal w/ removing it for now :)

**Pull request merged**